services/src/main/scala/cromwell/services/keyvalue/KeyValueServiceActor.scala

```diff
@@ -1,45 +1,56 @@
 package cromwell.services.keyvalue
 
 import akka.actor.{Actor, ActorRef}
-import com.typesafe.config.Config
-import cromwell.core.WorkflowId
+import cromwell.core.{JobKey, WorkflowId}
 import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
 import cromwell.services.keyvalue.KeyValueServiceActor._
 
 import scala.concurrent.{ExecutionContextExecutor, Future}
 import scala.util.{Failure, Success}
 
-
 object KeyValueServiceActor {
-  sealed  trait KvMessage
+  sealed trait KvMessage {
+    def key: ScopedKey
+  }
 
   sealed trait KvAction extends KvMessage with ServiceRegistryMessage {
     def serviceName = "KeyValue"
   }
-  case class KvJobKey(callFqn: String, callIndex: Option[Int], callAttempt: Int)
-  case class ScopedKey(workflowId: WorkflowId, jobKey: KvJobKey, key: String)
-  case class KvPut(pair: KvPair) extends KvAction
-  case class KvGet(key: ScopedKey) extends KvAction
+  final case class KvJobKey(callFqn: String, callIndex: Option[Int], callAttempt: Int)
+  object KvJobKey {
+    def apply(jobKey: JobKey): KvJobKey = KvJobKey(jobKey.scope.fullyQualifiedName, jobKey.index, jobKey.attempt)
+  }
+
+  final case class ScopedKey(workflowId: WorkflowId, jobKey: KvJobKey, key: String)
+  final case class KvPut(pair: KvPair) extends KvAction {
+    override def key = pair.key
+  }
+  final case class KvGet(key: ScopedKey) extends KvAction
 
-  sealed trait KvResponse extends KvMessage
-  case class KvPair(key: ScopedKey, value: Option[String]) extends KvResponse
-  case class KvFailure(action: KvAction, failure: Throwable) extends KvResponse
-  case class KvKeyLookupFailed(action: KvGet) extends KvResponse
-  case class KvPutSuccess(action: KvPut) extends KvResponse
+  sealed trait KvResponse extends KvMessage {
+    def key: ScopedKey
```

Comment:
Consider this request completely ignorable.

I find for these sorts trivial overrides that it winds up looking neater to handle the different values with a match in the trait. definitely not in cases where there's any sort of logic but this would IMO be a good use for it.

e.g. 

     def key = this match {
         case x: KvFailure => action.key
         case z: Foo => blah
         .....
     }